README.md

```diff
@@ -1,13 +1,69 @@
+</a>
+<img src="https://img.shields.io/badge/os-Mac%20OS%20X-green.svg?style=flat" alt="Mac OS X">
+<img src="https://img.shields.io/badge/os-linux-green.svg?style=flat" alt="Linux">
+<img src="https://img.shields.io/badge/license-Apache2-blue.svg?style=flat" alt="Apache 2">
+<a href="http://swift-at-ibm-slack.mybluemix.net/">
+<img src="http://swift-at-ibm-slack.mybluemix.net/badge.svg" alt="Slack Status">
+</a>
+</p>
+
 # Kitura-TemplateEngine
-![macOS](https://img.shields.io/badge/os-macOS-green.svg?style=flat)
-![Linux](https://img.shields.io/badge/os-linux-green.svg?style=flat)
-![Apache 2](https://img.shields.io/badge/license-Apache2-blue.svg?style=flat)
+Template engine abstraction layer
+
+## Summary
+Kitura-TemplateEngine provides a TemplateEngine protocol to unify the APIs of multiple template engines and integrate them with Kitura's content generation APIs. At runtime, the template engine replaces variables in a template file with actual values, and transforms the template into an HTML file sent to the client. This approach makes it easier to design an HTML page with integrated swift code.
+
+Kitura-TemplateEngine is easy to learn, consumable framework that comes with a set of [implemented plugins](#list-of-plugins).
+
+## Examples
+
+A Kitura-StencilTemplateEngine tutorial on adding a website front end to a food tracker application is available [here](https://github.com/IBM/FoodTrackerBackend/blob/master/StencilFrontend.md).
+
+The following code examples will use the [Kitura-StencilTemplateEngine](https://github.com/IBM-Swift/Kitura-StencilTemplateEngine), however, because this is an abstraction layer, Stencil could be substituted for any other supported template engine.
+
+The following code initializes a Stencil template engine and adds it to the [Kitura](https://github.com/IBM-Swift/Kitura) router.
+The template engine is added which will render files with its default file extensions (so for StencilTemplateEngine() this would be .stencil files).
+```swift
+router.add(templateEngine: StencilTemplateEngine())
+```
 
-## Description
-The Kitura Template Engine protocol. Implemented by Templating Engines in order to
-integrate with Kitura's content generation APIs.
+The following example specifies additional file extensions so the template engine will be used to render a file with the file extension ".example".  Because useDefaultFileExtension was set to false, the default file extentions (.stencil for StencilTemplateEngine) will not be rendered with this template engine.
 
-Inspired by http://expressjs.com/en/guide/using-template-engines.html.
+```swift
+router.add(templateEngine: StencilTemplateEngine(), forFileExtensions: [".example"], useDefaultFileExtension: false)
+```
+
+The following code will set the default template engine for the router. If there are any file extensions used which do not match to any template engines added to the router, then the router will render the file using this default template engine.
```

Comment:
If there are any files with file extensions which do not match to any of the template engines which have been added to the router, the router will render the file using the default template engine. You can set the default template engine for the router as follows: